R/standardize_MAE.R

```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
   }
   rd
 }
+
+
+#' Set Unique Parental Identifiers
+#'
+#' This function sets the `CellLineName` field in 
+#' `colData` to be unique by appending the `clid` in parentheses for duplicates.
+#'
+#' @param se A SummarizedExperiment object.
+#' @return A SummarizedExperiment object with unique `CellLineName` in `colData`.
+#' @examples
+#' se <- SummarizedExperiment::SummarizedExperiment(
+#'   assays = list(counts = matrix(1:4, ncol = 2)),
+#'   colData = S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C2"))
+#' )
+#' se <- set_unique_cl_names(se)
```

Comment:
It is 100% sure that `colData` has only unique records? That such scenario is impossible?
```
> se <- SummarizedExperiment::SummarizedExperiment(
   assays = list(counts = matrix(1:4, ncol = 2)),
   colData = S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C1"))
 )

SummarizedExperiment::colData(se)
DataFrame with 2 rows and 2 columns
  CellLineName        clid
   <character> <character>
1          ID1          C1
2          ID1          C1

se <- set_unique_cl_names(se)
SummarizedExperiment::colData(se)

DataFrame with 2 rows and 2 columns
  CellLineName        clid
   <character> <character>
1     ID1 (C1)          C1
2     ID1 (C1)          C1
```

Reply:
it results from our data model

Reply:
I had to ask. 😉 

✔️